lib/index.js

```diff
@@ -23,11 +23,11 @@ exports.clone = function (obj, seen) {
         return obj;
     }
 
-    seen = seen || { orig: [], copy: [] };
+    seen = seen || new Map();
 
-    const lookup = seen.orig.indexOf(obj);
-    if (lookup !== -1) {
-        return seen.copy[lookup];
+    const lookup = seen.get(obj);
+    if (lookup) {
```

Comment:
nah, the check at the top of the method doesn't allow you to attempt to clone something that could be falsey. should be fine just like this.